presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2886,7 +2919,19 @@ && isOrderBasedExecutionEnabled(session)) {
                 streamPartitionColumns,
                 discretePredicates,
                 localPropertyBuilder.build(),
-                Optional.of(hiveLayoutHandle.getRemainingPredicate()));
+                Optional.of(hiveLayoutHandle.getRemainingPredicate()),
+                canReplicatedReads);
+    }
+
+    protected boolean canPerformReplicatedReads(SchemaTableName tableName, HiveTableLayoutHandle hiveLayoutHandle, ConnectorSession session)
+    {
+        String tablePath = hiveLayoutHandle.getTablePath();
+        // temporary table may not have physical location
+        if (tablePath.isEmpty()) {
```

Comment:
Can you explain why this check is required?

Reply:
if this is a temporary table, there's no physical location, which does not satisfy for replicated reads execution